### DISCUSSION TITLE
Zero checksum support for SCTP

### DISCUSSION THREAD

**Discussion by voluntas** [paullouisageneau/libdatachannel#1012] [Ideas]
https://datatracker.ietf.org/doc/html/draft-ietf-tsvwg-sctp-zero-checksum-04

I would like to add to libdatachannel an RFC Draft support for zero checksum for SCTP.

WebRTC does not need to calculate the SCTP checksum (CRC32c) since SCTP works over DTLS.

Also supported by usrsctp
https://github.com/sctplab/usrsctp/commit/ca70d3af67059294eea4a4276ae17acdcff7c090

Support is also about to be added to dcsctp in libwebrtc
https://bugs.chromium.org/p/webrtc/issues/detail?id=14997&sort=-modified

Are there any negatives to adding this one?

**Comment by paullouisageneau** [comment]
libdatachannel already has an optimization not to verify SCTP checksums by leveraging CRC32 offloading as it's indeed unnecessary over DTLS:
https://github.com/paullouisageneau/libdatachannel/blob/5145ba8f4047a162838a553291a8e4d559588c50/src/impl/sctptransport.cpp#L105

I see no issue with enabling `SCTP_ACCEPT_ZERO_CHECKSUM`, it should provide a significant performance increase as computing CRC32 is the second biggest CPU cost after encryption.

**Comment by voluntas** [comment]
Thanks for the quick response!

I see that there was already a mechanism in place to not calculate the checksum of SCTP packets received by usrsctp.

If that is the case, it seems that there is no problem to enable the function to set the checksum of SCTP packets to 0 if the Zero Checksum Acceptable Chunk Parameter handshake is successful. Now let's consider a Pull-Request!
